contracts/utils/generalizedConvexStratDeployment.js

```diff
@@ -52,10 +54,19 @@ module.exports = ({
         dConvexTokenMetaStrategyProxy.address
       );
 
-      // 2. Deploy new implementation
-      const dConvexTokenMetaStrategyImpl = await deployWithConfirmation(
-        "ConvexGeneralizedMetaStrategy"
-      );
+      let convexTokenMetaStrategyImpl;
+
+      if (deployStrategyImplementation) {
```

Comment:
The thing you got back after a deploy command is the transaction, I think - a different kind of object than what you got from `ethers.getContract`, which is why we always made the distinction between them, with one of them being prepended with a `d`.

This code sets `convexTokenMetaStrategyImpl` with either kind. 

It looks like only the address is used? In which case perhaps we call this variable `implementationAddress`?

Or if we need to use it more, we could do the deploy in the if statement, then always ethers.getContract the contract regardless of if we have deployed it or not.

Reply:
Yup good point it is a different object, both seem to expose the address property. Have implemented optimisation you suggest: https://github.com/OriginProtocol/origin-dollar/pull/1002/commits/1301af1f2b690c40bacfcce948c3c70a75611d65